underpants.go

```diff
@@ -111,6 +127,7 @@ func (u *user) encode(key []byte) (string, error) {
   if err := json.NewEncoder(w).Encode(u); err != nil {
     return "", err
   }
+  w.Close()
```

Comment:
Sorry, you're right. I didn't expand the diff to see the bottom of the function. You need this thing closed well before you exit. Ignore what I said.